src/lib/AsyncTypeaheadInput.tsx

```diff
@@ -86,6 +89,22 @@ const AsyncTypeaheadInput = <T extends FieldValues>(props: AsyncTypeaheadProps<T
                 setIsLoading(false);
               })();
             }}
+            onBlur={() => {
+              if (options.length === 1) {
+                ref.current?.setState({
+                  selected: options,
+                });
+                setValue(name, typeof options[0] == "string" ? options[0] : options[0].value);
+              } else if (
+                options.length > 1 &&
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+                (!getValues(name) || convertTypeaheadOptionsToStringArray(options).includes(getValues(name)))
+              ) {
+                setError(name, { message: "Invalid Input" });
```

Comment:
@neotob @neotrow What do you thin kabout this? Would be the case the add a prop for custom message or a prop for a function that handle externally this case?

Reply:
if i understood it correctly, then this code is to auto-select an item if there is only 1 match right?
i would at least add a feature flag for it, so its not directly active for everyone updating (else it would be somehow a breaking change as it will change the behavior by just updating the package)
also with that logic, a async typeahed will always get mandatory correct? so not possible for having it optional in a from? (which would also be a breaking change)

Reply:
@manni497 mhm ok ok, i think if its really seen as a bugfix, then there is also no need for a flag to other way round to disable it, right? 🤔

ah okay i see the point, so if you search for something to check if its in the list and you just leave the typed text, without choosing anything, then it will always show invalid.. is there a way to already show items with just clicking in it? or you always have to type at least one character? just to make sure this case is also handled and not suddenly changed from optional to mandatory.